SPEC.md

```diff
@@ -215,11 +249,26 @@ An example trace:
 
 ```
 
-## Versioning
+### Resource restrictions
+
+In order to ensure analysis runs reliably across a variety of systems, Engines
+must conform to some basic resource restrictions:
+
+* The Docker image for an Engine must not exceed 512MB, including all layers.
+* The combined total RSS memory usage by all processes within the Docker container must not exceed 1GB at any time.
+* All Engines must complete and exit within 10 minutes.
 
-This specification is versioned. The current version is [in the repository](https://github.com/codeclimate/spec/blob/master/VERSION). Engines declare the version of the specification they are compatible with in the manifest file, described below. This project follows [Semantic Versioning](http://semver.org/).
+### Security restrictions
 
-## Engine Specification File
+Engines run in a secured runtime environment, within container-based virtualization
+provided by Docker.
+
+* The root filesystem (`/`) is mounted read-only. A `/tmp` volume is mounted read-write for temporary file storage during the engine run.
+* Engines run with no network access (`--net=none` in Docker). They must not rely on making any external network calls.
+* Engines run with the minimal set of Linux capabilities (`--cap-drop all` in Docker)
+* Engines are always run as a user `app` with UID and GID of 9000, and never `root`.
+
+## Engine specification file
```

Comment:
I don't feel super strongly either, it was just something that stuck out to me. Since this pre-existed I at least wouldn't worry about it for now.